### PR TITLE
Prepare text changes for non java files; several improvements

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: Valeriy Svydenko <vsvydenk@redhat.com>
- Make it possible to provide refactoring's changes for non Java fiels
- Added several improvements for bug fixing.

### CODE REVIEW

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java [hunk 1]
```diff
@@ -116,8 +112,18 @@ public static void convertChanges(Change change, CheWorkspaceEdit edit) throws C
...
+    if (change instanceof CompositeChange) {
+      for (Change child : ((CompositeChange) change).getChildren()) {
+        convertCompositeChange(child, edit, pm);
+      }
+    }
+    if (change instanceof TextFileChange) {
+      convertTextFileChange((TextFileChange) change, edit, pm);
+      return;
+    }
+
```
Comment (tsmaeder) [inline org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java]: Btw: if modifiedElement is null, it's not instanceof IJavaElement. The null check can go

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java [hunk 2]
```diff
@@ -116,8 +112,18 @@ public static void convertChanges(Change change, CheWorkspaceEdit edit) throws C
     return result;
   }
 
-  private static void convertCompositeChange(Change change, CheWorkspaceEdit edit)
-      throws CoreException {
+  private static void convertCompositeChange(
```
Comment (tsmaeder) [inline org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java]: If the method is called "convertCompositeChange" it should take a "CompositeChange". Do the type dispatching in one method and when you know the type, dispatch to typesafe methods.

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java (hunk 1 above)
Comment (tsmaeder) [reply org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java]: Does a ResourceChange have a modified element and does it have to be a IJavaElement? If not, the modifiedElement stuff needs to go inside the "if (change instanceof TextChange)".

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/FileTextEditConverter.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  public FileTextEditConverter(IDocument document, TextEdit edit) {
+    this.source = edit;
+    this.converted = new ArrayList<>();
+    if (document == null) {
+      throw new IllegalArgumentException("Document can not be null");
+    }
+    this.document = document;
+  }
+
+  public List<org.eclipse.lsp4j.TextEdit> convert() {
```
Comment (tsmaeder) [inline org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/FileTextEditConverter.java]: What if someone calls this method twice? I ususally solve the problem by making a public static method 
```
public static List<TextEdit> convert(document, edit) 
```
which creates an instance of FileTextEditConverter and calls convert on that. Then the constructor and the non-static convert method can be private. This way, you can't misuse the API.

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java (hunk 1 above)
Comment (svor) [reply org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java]: fixed

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java (hunk 2 above)
Comment (svor) [reply org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/ChangeUtil.java]: fixed

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/FileTextEditConverter.java (hunk 3 above)
Comment (svor) [reply org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/FileTextEditConverter.java]: fixed, thx!

**Pull request merged**